src/backend/backend_concept.h

```diff
@@ -1,34 +1,45 @@
 #pragma once
 
 #ifdef __cpp_concepts
-#  include <cstddef>
+#  include "../ds/address.h"
 #  include "../ds/concept.h"
 #  include "../pal/pal_concept.h"
-#  include "../ds/address.h"
+
+#  include <cstddef>
 namespace snmalloc
 {
+  template<typename BackendMetadata>
   class MetaEntry;
 
   /**
    * The core of the static pagemap accessor interface: {get,set}_metadata.
    *
-   * get_metadata takes a bool-ean template parameter indicating whether it may
+   * get_metadata takes a boolean template parameter indicating whether it may
    * be accessing memory that is not known to be committed.
    */
   template<typename Meta>
-  concept ConceptBackendMeta =
-    requires(
-      address_t addr,
-      size_t sz,
-      const MetaEntry& t)
+  concept ConceptBackendMeta = requires(
+    address_t addr, size_t sz, const typename Meta::Entry& t)
   {
-    { Meta::set_metaentry(addr, sz, t) } -> ConceptSame<void>;
+    {
+      Meta::template get_metaentry_mut<true>(addr)
```

Comment:
Is get_metaentry_mut used by the front-end? My understanding was that these are the pagemap accessors that any backend needs to implement and sandbox implementations don't normally implement get_metaentry_mut right?

Reply:
It's used by LargeBuddyRange, so you need it to exist if you use LargeBuddyRange.  I should probably separate it into a separate concept.

Reply:
Hmm, that is very useful to be aware of, but I am not sure if it can be captured in a concept, since backends are not required to use LargeBuddyRange, it is an internal implementation decision they make.

Reply:
Done in the latest version.

Reply:
I've pulled this out into a separate concept which LargeBuddyRange uses but nothing else.